File: examples/python/predict.py
import numpy as np
import wam


def predict_once(model_path, library_path, images, state, instruction):
    with wam.Pipeline.load(
        model_path,
        library=library_path,
        runtime_config=wam.RuntimeConfig(backend="cuda", precision="bf16"),
        session_config=wam.SessionConfig(random_seed=0),
    ) as pipeline:
        return pipeline.predict(images, np.asarray(state, dtype=np.float32),
                                instruction=instruction).action

File: examples/cpp/predict.cpp
#include "wam/wam.h"

// Application-owned storage referenced by Observation must outlive predict().
wam::Prediction predict_once(const std::string & artifact,
                             const wam::Observation & observation) {
    wam::Pipeline pipeline = wam::Pipeline::load(artifact);
    return pipeline.predict(observation);
}
